src/core/net/dns_headers.hpp

```diff
@@ -493,6 +493,136 @@ class Name
         kMaxEncodedLength = 255, ///< Max length of an encoded name.
     };
 
+    /**
+     * This enumeration represent the name type.
+     *
+     */
+    enum Type : uint8_t
+    {
+        kTypeEmpty,   ///< The name is empty (not specified).
+        kTypeString,  ///< The name is given as a string (dot '.' separated sequence of labels').
+        kTypeMessage, ///< The name is specified from a message at a given offset (encoded in the message).
```

Comment:
```suggestion
        kTypeEncoded, ///< The name is specified from a message at a given offset (encoded in the message).
```

Reply:
I prefer `kTypeMessage` over `kTypeEncoded` sicne it indicating that name is present in a message. `Encoded` may mean that it's is a buffer/array (encoded).